### PR TITLE
Refactor render plugins

### PR DESCRIPTION
**Pull request by pope1838**

### CODE REVIEW

File: amf-api-contract/shared/src/main/scala/amf/apicontract/client/scala/model/domain/EndPoint.scala [hunk 1]
```diff
@@ -76,8 +76,13 @@ class EndPoint(override val fields: Fields, override val annotations: Annotation
   override def meta: EndPointModel.type = EndPointModel
 
   /** Value , path + field value that is used to compose the id when the object its adopted */
-  private[amf] override def componentId: String = "/end-points/" + path.value().urlComponentEncoded
-  override def nameField: Field                 = Name
+  private[amf] override def componentId: String = "/end-points/" + {
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/client/scala/model/domain/EndPoint.scala:79]: @AgustinBettati 👀  final-id

File: amf-api-contract/shared/src/main/scala/amf/apicontract/client/scala/model/domain/Operation.scala [hunk 2]
```diff
@@ -102,9 +102,15 @@ case class Operation(fields: Fields, annotations: Annotations)
   override def meta: OperationModel.type = OperationModel
 
   /** Value , path + field value that is used to compose the id when the object its adopted */
-  private[amf] override def componentId: String =
-    "/" + method.option().getOrElse("default-operation").urlComponentEncoded
-  override def nameField: Field = Name
+  private[amf] override def componentId: String = {
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/client/scala/model/domain/Operation.scala:105]: @AgustinBettati 👀 final-id

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/ValidationTransformationPipeline.scala [hunk 3]
```diff
@@ -37,6 +32,7 @@ object ValidationTransformationPipeline {
     val pipeline = profile match {
       case Oas30Profile   => Oas30ValidationTransformationPipeline()
       case Async20Profile => Async20CachePipeline()
+      //case GrpcProfile    => GrpcTransformationPipeline()
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/ValidationTransformationPipeline.scala]: Erase, not comment.

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/ExtendsResolutionStage.scala [hunk 4]
```diff
@@ -198,7 +194,14 @@ class ExtendsResolutionStage(profile: ProfileName, val keepEditingInfo: Boolean,
         .getOrElse("")
     }
 
-    private def resourcePath(endPoint: EndPoint) = endPoint.path.value().replaceAll("\\{ext\\}", "")
+    private def resourcePath(endPoint: EndPoint) = {
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/ExtendsResolutionStage.scala]: could be replaced by fold

File: amf-cli/shared/src/main/scala/amf/cli/internal/commands/CmdLineParser.scala [hunk 5]
```diff
@@ -12,6 +12,8 @@ object CmdLineParser {
     Oas30Profile.profile == f ||
     Oas20Profile.profile == f ||
     AmfProfile.profile == f ||
+    GrpcProfile.profile == f ||
+    Proto3Profile.profile == f ||
```
Comment (tomsfernandez) [inline amf-cli/shared/src/main/scala/amf/cli/internal/commands/CmdLineParser.scala]: Wasn't this profile erased?

File: amf-cli/shared/src/main/scala/amf/cli/internal/commands/CmdLineParser.scala [hunk 6]
```diff
@@ -41,7 +43,7 @@ object CmdLineParser {
           success
         } else {
           failure(
-            s"Invalid value $f, values supported: '${Raml10.id}', '${Raml08.id}', '${Oas20.id}', '${Aml.id}', '${Amf.id}'")
+            s"Invalid value $f, values supported: '${Raml10.id}', '${Raml08.id}', '${Oas20.id}', '${Aml.id}', '${Amf.id}', '${Proto3.id}', ${Grpc.id}")
```
Comment (tomsfernandez) [inline amf-cli/shared/src/main/scala/amf/cli/internal/commands/CmdLineParser.scala]: Wasn't Proto3 spec erased?

File: build.sbt [hunk 7]
```diff
@@ -136,13 +137,119 @@ lazy val apiContractJS =
...
+    .disablePlugins(SonarPlugin, ScalaJsTypingsPlugin)
+
 /** **********************************************
   * AMF CLI
   * ********************************************* */
 lazy val cli = crossProject(JSPlatform, JVMPlatform)
   .settings(name := "amf-cli")
   .settings(fullRunTask(defaultProfilesGenerationTask, Compile, "amf.tasks.validations.ValidationProfileExporter"))
-  .dependsOn(apiContract)
+  .dependsOn(grpc)
```
Comment (tomsfernandez) [inline build.sbt:252]: Doesn't anyone depend on graphql?

File: build.sbt [hunk 8]
```diff
@@ -136,13 +137,119 @@ lazy val apiContractJS =
...
   .in(file("./amf-cli"))
```
Comment (tomsfernandez) [inline build.sbt:253]: Module dependencies is starting to get a bit weird. Grpc depending on apiContract (makes sense for the model but not for the rest.)

File: build.sbt (hunk 7 above)
Comment (pope1838) [reply build.sbt:252]: CLI will depend on graphql, but I didn't finish the minimun support yet

File: build.sbt (hunk 8 above)
Comment (pope1838) [reply build.sbt:253]: Totally, only need the model and other couple of classes, like contexts and similars.
We could thing on come back again with the idea of a api-core or api-common.

**Pull request merged**